pkg/edit/command_api.go

```diff
@@ -3,18 +3,34 @@ package edit
 import (
 	"src.elv.sh/pkg/cli/mode"
 	"src.elv.sh/pkg/eval"
+	"src.elv.sh/pkg/eval/vars"
 )
 
-//elv:fn command:start
+//elvdoc:fn command:start
 //
-// Starts the command mode.
+// Start command mode.
+
+//elvdoc:var command:-duration
+//
+// Duration, in seconds, of the most recent interactive command. This can be useful in your prompt
+// (left or right) to provide feedback on how long a command took to run. The initial value of this
+// variable is the time to evaluate your *~/.elvish/rc.elv* script before issuing the first prompt.
+
+var CommandDuration float64
```

Comment:
The `edit:command:` namespace is actually for the vi-like command-mode, putting the duration variable there would be a mistake.

I also commented that I prefer to just keep the hook; the variable can be easily emulated in Elvish code.

Reply:
Ahhh, yes, I simply latched onto the "command" namespace name. Part of the problem is this namespace has not documentation. See issue #971. I'll move it to the `edit` namespace pending further discussion about whether it is redundant given the callback.